### ISSUE TITLE
Possible bug: `user edit` removes user's `verified` flag?

### ISSUE THREAD

**Issue by tcompa** [fractal-analytics-platform/fractal-client#619]
To check:

1. Use credentials of the default admin (which is verified and superuser)
2. Check that the admin is verified and superuser
3. Find current user's ID
4. Edit the current user, by changing email and password via `fractal user edit`
5. Verify that the current user has the same flags as before (verified and superuser). If not, this is likely a bug.

**Comment by ychiucco** [comment]
```sh
$ poetry run fractal user whoami
{
  "id": 1,
  "email": "admin@fractal.xy",
  "is_active": true,
  "is_superuser": true,
  "is_verified": true,
  "slurm_user": null,
  "cache_dir": null,
  "username": "admin",
  "slurm_accounts": []
}

poetry run fractal user edit --new-email fake@email.com --new-password fakepassword 1
{
  "id": 1,
  "email": "fake@email.com",
  "is_active": true,
  "is_superuser": true,
  "is_verified": false,
  "slurm_user": null,
  "cache_dir": null,
  "username": "admin",
  "slurm_accounts": []
}
```
`is_verified` has changed.


But if we don't edit the email
```sh
$ poetry run fractal user show 1
{
  "id": 1,
  "email": "admin@fractal.xy",
  "is_active": true,
  "is_superuser": true,
  "is_verified": true,
  "slurm_user": null,
  "cache_dir": null,
  "username": "admin",
  "slurm_accounts": []
}
$ poetry run fractal user edit --new-password newpass 1
{
  "id": 1,
  "email": "admin@fractal.xy",
  "is_active": true,
  "is_superuser": true,
  "is_verified": true,
  "slurm_user": null,
  "cache_dir": null,
  "username": "admin",
  "slurm_accounts": []
}
```
`is_verified` is still true

**Comment by ychiucco** [comment]
https://github.com/fastapi-users/fastapi-users/blob/ae9f52474ba2c7baebeb923e4d03aea479765362/fastapi_users/manager.py#L674

**Comment by tcompa** [comment]
As per the last comment, this is the default behavior of fastapi-users. It looks funny in our case just because we are not associating "is_verified" to the email address.

Current working proposal, with @ychiucco:
1. If `fractal user edit` includes the `--new-email` field, and it includes either `--make-verified` or `--remove-verified`, the expected verified-related behavior is obtained.
2. If `fractal user edit` includes the `--new-email` field, but it does not include either `--make-verified` or `--remove-verified`, then CLI command fails before any API call.
3. If `fractal user edit` does not include the `--new-email` field, then the verified-related arguments are not treated in any special way.

TLDR: when updating a user's email, it becomes required to also specify if the user must be verified or not.

Note that we may have to make two API calls, within the CLI-command function, to make sure that the expected `is_verified` value is always set (ref https://github.com/fastapi-users/fastapi-users/blob/ae9f52474ba2c7baebeb923e4d03aea479765362/fastapi_users/manager.py#L674).
